actions/notify-slack/action.yml

```diff
@@ -0,0 +1,105 @@
+---
+
+name: 'Notify slack'
+description: 'Posts a slack message with the status of the pipeline'
+
+inputs:
+  application:
+    description: 'The friendly name for the application'
+    type: string
+    required: true
+  data:
+    description: 'The path to the file with data to pass to the render'
+    type: string
+    required: false
+    default: ''
+  environment:
+    description: |
+      The environment variables to pass to the render.  Each environment variable should be in format VAR=VAL with one on each line.
+      Example:
+        environment: |
+          FOO=bar
+          BAR=baz
+    type: string
+    required: false
+    default: ''
+  format:
+    description: 'The format the data file will be in'
+    type: string
+    required: false
+    default: ''
+  version:
+    description: 'The version that is being deployed'
+    type: string
+    default: ${{ github.ref }}
+  type:
+    description: 'The type of action that is being performed such as build or deployment'
+    type: string
+    required: true
+  template:
+    description: 'The path to the template file to render'
+    type: string
+    required: true
+  status:
+    description: 'The status of the deployment.  Options are started, success, and failure'
+    type: string
+    required: true
+  slack-webhook-url:
+    description: 'The slack webhook url to post the message to'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: 'Create data file'
+      id: data
+      env:
+        APPLICATION: ${{ inputs.application }}
+        ENVIRONMENT: ${{ inputs.environment }}
+        VERSION: ${{ inputs.version }}
+        TYPE: ${{ inputs.type }}
+        STATUS: ${{ inputs.status }}
+        ACTOR: ${{ github.actor }}
+        RUN_ID: ${{ github.run_id }}
+        REPOSITORY: ${{ github.repository }}
+        TEMPLATE: ${{ inputs.template }}
+        REF: ${{ github.ref_name }}
+      run: |
+        set -eo pipefail
+        TIMESTAMP="$(date +%s)"
+        {
+        echo '{'
+        echo ' "application": "'"$APPLICATION"'",'
+        echo ' "environment": "'"$ENVIRONMENT"'",'
+        echo ' "version": "'"$VERSION"'",'
+        echo ' "type": "'"$TYPE"'",'
+        echo ' "status": "'"$STATUS"'",'
+        echo ' "repository": "'"$REPOSITORY"'",'
+        echo ' "ref": "'"$REF"'",'
+        echo ' "timestamp": "'"$TIMESTAMP"'",'
+        echo ' "actor": "'"$ACTOR"'",'
+        echo ' "run_id": "'"$RUN_ID"'",'
+        echo ' "link": "'"$LINK"'",'
+        echo ' "message": "'"$MESSAGE"'",'
+        echo ' "emoji": "'"$EMOJI"'"'
+        echo '}'
+        } > data.json
+        echo "file=$PWD/data.json" >> $GITHUB_OUTPUT
+      shell: bash
+
```

Comment:
In turn, you will need to change the message.json.j2 to pull `env()` instead of a raw variable.  https://github.com/kolypto/j2cli#using-environment-variables